lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -142,13 +143,16 @@ def bulk(actions)
         stream_writer.write(as_json)
         batch_actions << action
       end
+
       stream_writer.close if http_compression
+
       logger.debug("Sending final bulk request for batch.",
                    :action_count => batch_actions.size,
                    :payload_size => stream_writer.pos,
                    :content_length => body_stream.size,
                    :batch_offset => (actions.size - batch_actions.size))
-      bulk_responses << bulk_send(body_stream, batch_actions) if body_stream.size > 0
+      bulk_responses << bulk_send(body_stream, batch_actions)
```

Comment:
IIRC this `if body_stream.size > 0` was a safeguard against sending zero-entry bulks. I haven't validated whether those are still possible.

Reply:
Yeah I accidentally removed it, put back in my previous commit.